src/content/posts/javascript/variables-javascript.mdx

```diff
@@ -0,0 +1,97 @@
+---
+title: Variables en JavaScript
+description: Aprende a declarar variables en JavaScript
+publishDate: 2025-07-28
+tags: ["JavaScript"]
+---
+
+import { Image } from "astro:assets";
+import YouTubeEmbed from "../../../components/YoutubeEmbed.astro";
+
+# Variables en JavaScript
+
+En JavaScript se pueden declarar variables de 4 maneras diferentes:
+
+* var (en desuso desde el 2015)
+* let
+* const
+* Automáticamente (no recomendado)
+
+## Ejemplo de Var 
+
+### var está en desuso desde el 2015, pero aún se puede usar. No es una buena práctica usarlo porque puede causar problemas de alcance (scope) y redeclaración.
+
+```javascript
+<script>
+  var x = 5;
+  var y = 6;
+  var z = x + y;
+</script>
+```
+En este caso la variable `x` se declara con el valor 5, la variable `y` con el valor 6 y la variable `z` es la suma de `x` y `y`, por lo que su valor será 11.
+
+## Ejemplo de Let
+
+```javascript
+<script>
+  let x = 5;
+  let y = 6;
+  let z = x + y;
+</script>
+```
+
+En este caso la variable `x` se declara con el valor 5, la variable `y` con el valor 6 y la variable `z` es la suma de `x` y `y`, por lo que su valor será 11.  
+La diferencia con `var` es que `let` tiene un alcance de bloque, lo que significa que solo está disponible dentro del bloque donde se declara.
+## Ejemplo de Const
+
+```javascript
+<script>
+  const x = 5;
+  const y = 6;
+  const z = x + y;
+</script>
+```
+
+En este caso la variable `x` se declara con el valor 5, la variable `y` con el valor 6 y la variable `z` es la suma de `x` y `y`, por lo que su valor será 11.  
+La diferencia con `let` es que `const` no se puede reasignar, es decir, una vez que se declara una variable con `const`, no se puede cambiar su valor. Esto es útil para declarar constantes que no deben cambiar durante la ejecución del programa.
+
+## Ejemplo mixto
+
+```javascript
+<script>
+  const price1 = 5;
+  const price2 = 6;
+  let total = price1 + price2;
+```
+En este caso, `price1` y `price2` son constantes que no cambiarán, mientras que `total` es una variable que puede cambiar si se recalcula. Esto es útil para mantener la integridad de los valores que no deben cambiar.
+
+## ¿Cuando usar let o const?
+
+1. Úselo siempre const si no se debe cambiar el valor.
+
+2. Úselo siempre const si no se debe cambiar el tipo (matrices y objetos)
+
+3. Úselo solo let si no lo puede manejar const
+
+4. Úselo solo var si DEBE admitir navegadores antiguos.
```

Comment:
Improve: mention scope of variables when using var to understand the implications and addition of let and const to the language